### PR TITLE
Fix #1001: Grey backdrop after switching workspace via gestures

### LINKED ISSUE #1001: Grey backdrop after switching workspace via gestures
@JoseExposito I think that you said you were never able to reproduce it

## What Happened
Using the workspace gestures by lightly swiping in opposing directions in quick succession causes the background to be replaced by the system grey backdrop (it doesn't actually switch workspace).

## Expected Behavior
Workspace switching should not create such bugs.

## Steps to Reproduce

1. Swipe to another workspace quickly, with only enough contact with the fingers to just about toggle the animation
2. Do it again in the other direction
3. Repeat until you see the bug

Video for referance:

https://user-images.githubusercontent.com/58219504/103150631-f1488080-476d-11eb-8e0a-0a0737095c5c.mp4
### Logs

Utterly useless:
```
Window manager warning: META_CURRENT_TIME used to choose focus window; focus window may not be correct.
Window manager warning: META_CURRENT_TIME used to choose focus window; focus window may not be correct.
Window manager warning: META_CURRENT_TIME used to choose focus window; focus window may not be correct.
Window manager warning: META_CURRENT_TIME used to choose focus window; focus window may not be correct.
Window manager warning: META_CURRENT_TIME used to choose focus window; focus window may not be correct.
Window manager warning: META_CURRENT_TIME used to choose focus window; focus window may not be correct.
Window manager warning: META_CURRENT_TIME used to choose focus window; focus window may not be correct.
Window manager warning: META_CURRENT_TIME used to choose focus window; focus window may not be correct.
Window manager warning: META_CURRENT_TIME used to choose focus window; focus window may not be correct.
Window manager warning: META_CURRENT_TIME used to choose focus window; focus window may not be correct.
 Window manager warning: META_CURRENT_TIME used to choose focus window; focus window may not be correct.
Window manager warning: META_CURRENT_TIME used to choose focus window; focus window may not be correct.
  Window manager warning: META_CURRENT_TIME used to choose focus window; focus window may not be correct.
Window manager warning: META_CURRENT_TIME used to choose focus window; focus window may not be correct.
Window manager warning: META_CURRENT_TIME used to choose focus window; focus window may not be correct.
Window manager warning: META_CURRENT_TIME used to choose focus window; focus window may not be correct.
Window manager warning: META_CURRENT_TIME used to choose focus window; focus window may not be correct.
Window manager warning: META_CURRENT_TIME used to choose focus window; focus window may not be correct.
 Window manager warning: META_CURRENT_TIME used to choose focus window; focus window may not be correct.
Window manager warning: META_CURRENT_TIME used to choose focus window; focus window may not be correct.
Window manager warning: META_CURRENT_TIME used to choose focus window; focus window may not be correct.
Window manager warning: META_CURRENT_TIME used to choose focus window; focus window may not be correct.
Window manager warning: META_CURRENT_TIME used to choose focus window; focus window may not be correct.
Window manager warning: META_CURRENT_TIME used to choose focus window; focus window may not be correct.
Window manager warning: META_CURRENT_TIME used to choose focus window; focus window may not be correct.
Window manager warning: META_CURRENT_TIME used to choose focus window; focus window may not be correct.
Window manager warning: META_CURRENT_TIME used to choose focus window; focus window may not be correct.
Window manager warning: META_CURRENT_TIME used to choose focus window; focus window may not be correct.
Window manager warning: META_CURRENT_TIME used to choose focus window; focus window may not be correct.
Window manager warning: META_CURRENT_TIME used to choose focus window; focus window may not be correct.
Window manager warning: META_CURRENT_TIME used to choose focus window; focus window may not be correct.
Window manager warning: META_CURRENT_TIME used to choose focus window; focus window may not be correct.
Window manager warning: META_CURRENT_TIME used to choose focus window; focus window may not be correct.
Window manager warning: META_CURRENT_TIME used to choose focus window; focus window may not be correct.
Window manager warning: META_CURRENT_TIME used to choose focus window; focus window may not be correct.
Window manager warning: META_CURRENT_TIME used to choose focus window; focus window may not be correct.
Window manager warning: META_CURRENT_TIME used to choose focus window; focus window may not be correct.
Window manager warning: META_CURRENT_TIME used to choose focus window; focus window may not be correct.
Window manager warning: META_CURRENT_TIME used to choose focus window; focus window may not be correct.
Window manager warning: META_CURRENT_TIME used to choose focus window; focus window may not be correct.
Window manager warning: META_CURRENT_TIME used to choose focus window; focus window may not be correct.
Window manager warning: META_CURRENT_TIME used to choose focus window; focus window may not be correct.
Window manager warning: META_CURRENT_TIME used to choose focus window; focus window may not be correct.
Window manager warning: META_CURRENT_TIME used to choose focus window; focus window may not be correct.
Window manager warning: META_CURRENT_TIME used to choose focus window; focus window may not be correct.
Window manager warning: META_CURRENT_TIME used to choose focus window; focus window may not be correct.
Window manager warning: META_CURRENT_TIME used to choose focus window; focus window may not be correct.
Window manager warning: META_CURRENT_TIME used to choose focus window; focus window may not be correct.
Window manager warning: META_CURRENT_TIME used to choose focus window; focus window may not be correct.
Window manager warning: META_CURRENT_TIME used to choose focus window; focus window may not be correct.
Window manager warning: META_CURRENT_TIME used to choose focus window; focus window may not be correct.
Window manager warning: META_CURRENT_TIME used to choose focus window; focus window may not be correct.
Window manager warning: META_CURRENT_TIME used to choose focus window; focus window may not be correct.
Window manager warning: META_CURRENT_TIME used to choose focus window; focus window may not be correct.
Window manager warning: META_CURRENT_TIME used to choose focus window; focus window may not be correct.
```

## Platform Information
elementary OS 6 Daily, Gala compiled from latest git

### PR DESCRIPTION
**Pull request by JoseExposito**
When the background actor was reparent from the `background_group` to the switch desktop container and reparent again in the `background_group` in the same X coordinate, for some reason, it became invisible.

It is a weird bug, because moving the background +/- 1 pixel made it visible.
In the old code:

```
background.x = background.steal_data<int> ("prev-x");
background.y = background.steal_data<int> ("prev-y");
background.x += 1; // This makes the background visible (?)
background.x -= 1; // It also works with "y" (??)
continue;
```

Obviously, the is not the fix of this PR :sweat_smile: Instead of reparenting the background actor and using the real wallpaper in the animation, I clone it and make it invisible. This change also simplifies the code a little bit.

Fix https://github.com/elementary/gala/issues/1001

### CODE REVIEW

File: src/WindowManager.vala [hunk 1]
```diff
@@ -1740,20 +1740,20 @@ namespace Gala {
...
             if (move_primary_only) {
-                wallpaper = background_group.get_child_at_index (primary);
-                wallpaper.set_data<int> ("prev-x", (int) wallpaper.x);
-                wallpaper.set_data<int> ("prev-y", (int) wallpaper.y);
-            } else
-                wallpaper = background_group;
-
-            windows.prepend (wallpaper);
-            parents.prepend (wallpaper.get_parent ());
+                var background = background_group.get_child_at_index (primary);
```
Comment (tintou) [inline src/WindowManager.vala:1743]:
```suggestion
                unownedvar background = background_group.get_child_at_index (primary);
```

File: src/WindowManager.vala [hunk 2]
```diff
@@ -1969,27 +1969,20 @@ namespace Gala {
             unowned Meta.Display display = get_display ();
             var active_workspace = display.get_workspace_manager ().get_active_workspace ();
 
+            // Show the real wallpaper again
+            var primary = display.get_primary_monitor ();
+            var move_primary_only = InternalUtils.workspaces_only_on_primary ();
+            if (move_primary_only) {
+                var background = background_group.get_child_at_index (primary);
```
Comment (tintou) [inline src/WindowManager.vala:1976]:
```suggestion
                unowned var background = background_group.get_child_at_index (primary);
```

**Comment by tintou** [comment]
Two tiny things and ready to merge :slightly_smiling_face:

**Comment by JoseExposito** [comment]
Thanks a lot for the code review @tintou , good catch!

**Review by tintou — APPROVED** [review]

**Pull request merged**